app/main.py

```diff
@@ -1,8 +1,20 @@
 class Person:
-    # write your code here
-    pass
+
+    people = {}   # class attribute
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name  # instance attribute
+        self.age = age  # instance attribute
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = [Person(person["name"], person["age"]) for person in people]
+    for person in people:
+        new_key = "wife" if "wife" in person else "husband"
+        if person.get(new_key):
+            setattr(
+                Person.people[person["name"]],
+                new_key, Person.people[person[new_key]]
```

Comment:
Ensure that `person[new_key]` exists in `Person.people` before accessing it. Otherwise, this could raise a KeyError if the referenced person is not already created.